porch/repository/pkg/git/testing_repo.go

```diff
@@ -183,37 +183,45 @@ func findFile(t *testing.T, tree *object.Tree, path string) *object.File {
 	return file
 }
 
-func findPackage(t *testing.T, revisions []repository.PackageRevision, name string) repository.PackageRevision {
+func findPackage(t *testing.T, revisions []repository.PackageRevision, name repo.PackageRevisionName) repository.PackageRevision {
 	for _, r := range revisions {
-		if r.Name() == name {
+		rev, err := r.GetPackageRevision()
+		if err != nil {
+			t.Fatalf("GetPackageRevision of %q failed: %v", r.Name(), err)
+		}
+		if rev.Spec.RepositoryName == name.Repository &&
```

Comment:
An advantage of `Key()` could be that I think we could then just do a `==` check against the two structs.

Reply:
done